### ISSUE TITLE
manual/javaGuide/main/templates/JavaCustomTemplateFormat.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#966]
- https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md
- diffs between [2.2.0/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md) and [2.3.5/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md](https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md    2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/javaGuide/main/templates/JavaCustomTemplateFormat.md   2014-07-28 12:26:38.000000000 +0900
@@ -1,3 +1,4 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Adding support for a custom format to the template engine

 The built-in template engine supports common template formats (HTML, XML, etc.) but you can easily add support for your own formats, if needed. This page summarizes the steps to follow to support a custom format.
@@ -21,13 +22,13 @@

 ## Implement a format

-Implement the `play.templates.Format<A>` interface that has the methods `A raw(String text)` and `A escape(String text)` that will be used to integrate static and dynamic template parts, respectively.
+Implement the `play.twirl.api.Format<A>` interface that has the methods `A raw(String text)` and `A escape(String text)` that will be used to integrate static and dynamic template parts, respectively.

-The type parameter `A` of the format defines the result type of the template rendering, e.g. `Html` for a HTML template. This type must be a subtype of the `play.templates.Appendable<A>` trait that defines how to concatenates parts together.
+The type parameter `A` of the format defines the result type of the template rendering, e.g. `Html` for a HTML template. This type must be a subtype of the `play.twirl.api.Appendable<A>` trait that defines how to concatenates parts together.

-For convenience, Play provides a `play.api.templates.BufferedContent<A>` abstract class that implements `play.templates.Appendable<A>` using a `StringBuilder` to build its result and that implements the `play.mvc.Content` interface so Play knows how to serialize it as an HTTP response body.
+For convenience, Play provides a `play.twirl.api.BufferedContent<A>` abstract class that implements `play.twirl.api.Appendable<A>` using a `StringBuilder` to build its result and that implements the `play.twirl.api.Content` interface so Play knows how to serialize it as an HTTP response body.

-In short, you need to write to classes: one defining the result (implementing `play.templates.Appendable<A>`) and one defining the text integration process (implementing `play.templates.Format<A>`). For instance, here is how the HTML format could be defined:
+In short, you need to write two classes: one defining the result (implementing `play.twirl.api.Appendable<A>`) and one defining the text integration process (implementing `play.twirl.api.Format<A>`). For instance, here is how the HTML format could be defined:

 '''java
 public class Html extends BufferedContent<Html> {
@@ -48,12 +49,12 @@

 ## Associate a file extension to the format

-The templates are compiled into a `.scala` files by the build process just before compiling the whole application sources. The `sbt.PlayKeys.templatesTypes` key is a sbt setting of type `Map[String, String]` defining the mapping between file extensions and template formats. For instance, if you want Play to use your onw HTML format implementation you have to write the following in your build file to associate the `.scala.html` files to your custom `my.HtmlFormat` format:
+The templates are compiled into a `.scala` files by the build process just before compiling the whole application sources. The `TwirlKeys.templateFormats` key is a sbt setting of type `Map[String, String]` defining the mapping between file extensions and template formats. For instance, if you want Play to use your own HTML format implementation you have to write the following in your build file to associate the `.scala.html` files to your custom `my.HtmlFormat` format:

 '''scala
-templatesTypes += ("html" -> "my.HtmlFormat.instance")
+TwirlKeys.templateFormats += ("html" -> "my.HtmlFormat.instance")
 '''

-Note that the right side of the arrow contains the fully qualified name of a static value of type `play.templates.Format<?>`.
+Note that the right side of the arrow contains the fully qualified name of a static value of type `play.twirl.api.Format<?>`.

-> **Next:** [[HTTP form submission and validation | JavaForms]]
\ No newline at end of file
+> **Next:** [[HTTP form submission and validation | JavaForms]]

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1123
